Distiller-server/src/main/java/com/mapr/distiller/server/recordtypes/SystemCpuRecord.java

```diff
@@ -73,6 +77,53 @@ public SystemCpuRecord() throws Exception {
 					this.cpu_softirq.add(
 					this.cpu_steal.add(
 					this.cpu_other))))))));
+			
+			//Now process single cpu lines
+			LinkedList<String[]> cpuLines = new LinkedList<String[]>();
```

Comment:
I've updated the code, see the new version.  I am still using LinkedList, but the calls to it are limited to add (which shouldn't require traversal as LinkedList maintains a pointer to the last element, I think), get(0) (which again, shouldn't require any traversal), and remove(0) (which also should require any traversal).

I chose not to use ArrayList because I don't know how big of an array I need until parsing the entire file.  Undersizing the ArrayList is expensive when resize happens, oversizing the ArrayList is wasteful and we already have a bit of memory churn in the code right now that is going to need to be addressed through optimizations at some point.